### PR TITLE
Provide access to enclosing type

### PR DESCRIPTION
**Pull request by timowest**

### CODE REVIEW

**Comment by Shredder121** [comment]
What do you think of adding `Type.isNested()`?

**Comment by Shredder121** [comment]
Or does it complicate the API?

**Comment by timowest** [comment]
Or `Type.isMember()`? That's similar to `Class.isMemberClass()`.

**Comment by Shredder121** [comment]
Yeah, I haven't thought of that. That sounds good to me.

File: src/main/java/com/mysema/codegen/model/SimpleType.java [hunk 1]
```diff
@@ -145,6 +146,16 @@ public Type getComponentType() {
     }
 
     @Override
+    public Type getEnclosingType() {
+        if (enclosingType == null && localName.contains(".")) {
```
Comment (Shredder121) [inline src/main/java/com/mysema/codegen/model/SimpleType.java]: maybe change `localName.contains(".")` to `memberClass`?

**Pull request merged**